### ISSUE TITLE
MA Part A AGI may double-count capital losses when LTCG >= STCG loss

### ISSUE THREAD

**Issue by PavelMakarchuk** [PolicyEngine/policyengine-us#6991]
## Summary

Massachusetts Part A AGI calculation may double-count short-term capital losses when:
1. Short-term capital losses exist
2. Positive long-term capital gains exist (≥ STCG loss amount)
3. Interest and/or dividends exist

This was discovered during review of PR #6969 (which correctly fixes Part C).

## Root Cause

In `ma_part_a_agi.py`, the calculation uses the **full** STCG loss amount against interest/dividends without considering how much was already "consumed" by offsetting LTCG in Part C.

```python
short_term_capital_loss = max_(0, -short_term_capital_gains)
# ...
short_term_loss_against_interest_dividends = min_(
    interest_dividends_deduction_cap,  # $2,000
    min_(
        interest_and_dividends,
        short_term_capital_loss,  # Uses FULL loss, not remaining after Part C
    ),
)
```

## Example Scenario

**Inputs:**
- Short-term capital gains: -$10,000 (loss)
- Long-term capital gains: +$10,000 (gain)
- Dividends: $5,000

**Per MA TIR 02-21 rules (correct behavior):**
1. STCG loss offsets LTCG: $10,000 vs $10,000 → $0 remaining loss
2. Remaining loss offsets dividends: $0 → $5,000 dividends remain taxable

**Current code behavior:**
- Part C: $10,000 LTCG - $10,000 loss = $0 ✅
- Part A: $5,000 dividends - $2,000 (from $10,000 loss, capped) = $3,000 ❌

**Result:** $12,000 of deductions claimed from $10,000 of losses

## Suggested Fix

Part A should calculate the remaining STCG loss after Part C consumption:

```python
# STCG losses used in Part C to offset LTCG
stcg_loss_used_in_part_c = min_(short_term_capital_loss, max_(0, long_term_capital_gains))
remaining_short_term_loss = short_term_capital_loss - stcg_loss_used_in_part_c

# Then use remaining_short_term_loss instead of short_term_capital_loss
short_term_loss_against_interest_dividends = min_(
    interest_dividends_deduction_cap,
    min_(
        interest_and_dividends,
        remaining_short_term_loss,  # Only use what's left after Part C
    ),
)
```

## Test Case

```yaml
- name: STCG loss fully consumed by LTCG should not offset dividends
  period: 2024
  input:
    short_term_capital_gains: -10_000
    long_term_capital_gains: 10_000
    dividend_income: 5_000
    state_code: MA
  output:
    ma_part_c_gross_income: 0  # LTCG offset by STCG loss
    ma_part_a_agi: 5_000  # Full dividends taxable (no remaining loss)
```

## References

- [MA TIR 02-21](https://www.mass.gov/technical-information-release/tir-02-21-capital-gains-and-losses-massachusetts-tax-law-changes) - Loss offset hierarchy
- [MA Schedule B Instructions](https://www.mass.gov/doc/2024-schedule-b-interest-dividends-and-certain-capital-gains-and-losses/download)
- Related PR: #6969

## Files to Fix

- `policyengine_us/variables/gov/states/ma/tax/income/adjusted_gross_income/ma_part_a_agi.py`
